alpine-nix-rails-nginx/etc/nginx/nginx.conf

```diff
@@ -0,0 +1,103 @@
+env PATH;
+env TZ=utc;
+
+user nginx nogroup;
+
+worker_processes auto;
+worker_priority  15;  # limit worst-case resource utilization to 25%
+
+events {
+  worker_connections 1024;
+}
+
+http {
+  # Do not leak application information
+  server_tokens off;
+  more_clear_headers Server X-Powered-By;
+
+  sendfile    on;
+  tcp_nopush  on;
+  tcp_nodelay on;
+
+  keepalive_timeout 90;
+
+  include      /nix/var/nix/profiles/default/conf/mime.types;
+  default_type application/octet-stream;
+
+  types {
+    image/icns            icns;
+    application/font-sfnt ttf otf;
+  }
+
+  source_charset utf-8;
+  charset        utf-8;
+  charset_types  text/plain text/css application/javascript application/json text/xml application/xml application/xml+rss image/svg+xml;
+
+  gzip              on;
+  gzip_vary         on;
+  gzip_static       on;
+  gzip_proxied      any;
+  gzip_disable      "msie6";
+  gzip_comp_level   9;
+  gzip_http_version 1.1;
+  gzip_types        text/plain text/css application/javascript application/json text/xml application/xml application/xml+rss image/svg+xml image/icns image/x-icon application/font-sfnt application/vnd.ms-fontobject;
+
+  real_ip_header    X-Forwarded-For;
+  set_real_ip_from  10.0.0.0/8;
+  set_real_ip_from  172.16.0.0/12;
+  set_real_ip_from  192.168.0.0/16;
+  real_ip_recursive on;
+
+  limit_req_zone $binary_remote_addr zone=dynamic:10m rate=10r/s;
+
+  proxy_temp_path        /var/cache/nginx/tmp;
+  proxy_cache_path       /var/cache/nginx/proxy keys_zone=cacheable:10m;
+  proxy_cache_use_stale  error timeout invalid_header updating http_500 http_502 http_503 http_504;
+  proxy_cache_lock       on;
+  proxy_cache_revalidate on;
+  proxy_intercept_errors on;
+  proxy_http_version     1.1;
+  proxy_set_header       Host              $host;
+  proxy_set_header       X-Forwarded-For   $proxy_add_x_forwarded_for;
+  proxy_set_header       X-Forwarded-Proto $scheme;
+  proxy_set_header       X-Real-IP         $remote_addr;
+  proxy_set_header       X-Request-Start   "t=${msec}";
+  proxy_set_header       Connection        "";
+
+  merge_slashes off;
+
+  if_modified_since before;
+
+  more_set_headers "X-Content-Type-Options: nosniff";
+  more_set_headers "X-Frame-Options: deny";
+  more_set_headers "X-XSS-Protection: 1; mode=block";
```

Comment:
I wounder if the ELB will see this and "freak out" when the backend does not offer HTTPS. (AFAIK it does not, just double checking).